common/component/kafka/kafka.go

```diff
@@ -182,19 +188,32 @@ func (k *Kafka) Init(ctx context.Context, metadata map[string]string) error {
 		// already handled in updateTLSConfig
 	case awsIAMAuthType:
 		k.logger.Info("Configuring AWS IAM authentication")
-		err = updateAWSIAMAuthInfo(k.internalContext, config, meta)
+		kafkaIAM, validateErr := k.ValidateAWS(metadata)
+		if validateErr != nil {
+			return fmt.Errorf("failed to validate AWS IAM authentication fields: %w", validateErr)
+		}
+		opts := awsAuth.Options{
+			Logger:        k.logger,
+			Properties:    metadata,
+			Region:        kafkaIAM.Region,
+			Endpoint:      "",
+			AccessKey:     kafkaIAM.AccessKey,
+			SecretKey:     kafkaIAM.SecretKey,
+			SessionToken:  kafkaIAM.SessionToken,
+			AssumeRoleARN: kafkaIAM.IamRoleArn,
+			SessionName:   kafkaIAM.StsSessionName,
+		}
+		var provider awsAuth.Provider
+		provider, err = awsAuth.NewProvider(ctx, opts, awsAuth.GetConfig(opts))
 		if err != nil {
 			return err
 		}
+		k.awsAuthProvider = provider
 	}
 
 	k.config = config
 	sarama.Logger = SaramaLogBridge{daprLogger: k.logger}
-
-	k.producer, err = getSyncProducer(*k.config, k.brokers, meta.MaxMessageBytes)
```

Comment:
removing this initialization of producer and consumer and moving it to the lazily loaded function `latestClients` changes the semantics of the component init

now we would not validate the kafka connection until a publish or subscribe request is executed. I think this should be changed and the `clients` field should be initialized here